check-webkit-style should skip *.py files under LayoutTests/imported directory

fujii opened the pull request:
#### 1b97d0c6ae2e7e17b66ab0161fb518fd9b1fde05
<pre>
check-webkit-style should skip *.py files under LayoutTests/imported directory
<a href="https://bugs.webkit.org/show_bug.cgi?id=274407">https://bugs.webkit.org/show_bug.cgi?id=274407</a>

Reviewed by Jonathan Bedard.

check-webkit-style reported errors for *.py files under
web-platform-tests directory. It should ignore *.py files under
LayoutTests/imported directory.

_NEVER_SKIPPED_FILES contained a regexp
&quot;re.compile(&apos;(?!WebDriverTests).{0,14}.*.py$&apos;)&quot; which matches all py
files under all directory but `WebDriverTests`.

Changed it to a regexp matches *.py files udner `LayoutTests` directory
but not followed by `imported`.

* Tools/Scripts/webkitpy/style/checker.py:
* Tools/Scripts/webkitpy/style/checker_unittest.py:
(CheckerDispatcherSkipTest.test_should_skip_without_warning__true):

Canonical link: <a href="https://commits.webkit.org/284464@main">https://commits.webkit.org/284464@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/27635a31234b080f8021f901f5260d08b2cc0536

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/69371 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/48771 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/22044 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/73454 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/20529 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/71488 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/56572 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/20380 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/55173 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/13635 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/72437 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/44494 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/59892 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/35651 "Passed tests") | 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/6/builds/68880 "Passed tests") | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/41160 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/17322 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/18906 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/63102 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/17667 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/75165 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/13353 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/16890 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/62839 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/13392 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/59975 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/62745 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/10763 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/4376 "Passed tests") | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/10605 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/44575 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/45649 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/46844 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/45390 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

karlcow reviewed (commented):
Approving from a non-reviewer (so as a comment.)
Thanks. I let @JonWBedard gives the final approval.

karlcow commented on Tools/Scripts/webkitpy/style/checker.py:
```diff
@@ -459,7 +459,7 @@
     re.compile('.*TestExpectations.json$'),
     # Avoid imported WebDriverTests python machinery
-    re.compile('(?!WebDriverTests).{0,14}.*.py$'),
+    re.compile('^LayoutTests.(?!imported).*\\.py$'),
```
There are a lot of python scripts into `WebDriverTests`. Are we sure we want to exclude them, instead of adding LayoutTests too.

karlcow replied:
```
_NEVER_SKIPPED_FILES contained a regexp
"re.compile('(?!WebDriverTests).{0,14}.*.py$')" which matches all py
files under all directory but `WebDriverTests`.
```

Ah understood.

karlcow reviewed (commented):
One thing to make sure.

fujii committed:
check-webkit-style should skip *.py files under LayoutTests/imported directory
https://bugs.webkit.org/show_bug.cgi?id=274407

Reviewed by Jonathan Bedard.

check-webkit-style reported errors for *.py files under
web-platform-tests directory. It should ignore *.py files under
LayoutTests/imported directory.

_NEVER_SKIPPED_FILES contained a regexp
"re.compile('(?!WebDriverTests).{0,14}.*.py$')" which matches all py
files under all directory but `WebDriverTests`.

Changed it to a regexp matches *.py files udner `LayoutTests` directory
but not followed by `imported`.

* Tools/Scripts/webkitpy/style/checker.py:
* Tools/Scripts/webkitpy/style/checker_unittest.py:
(CheckerDispatcherSkipTest.test_should_skip_without_warning__true):

Canonical link: https://commits.webkit.org/284464@main
